internal/manifest/oci_index_test.go

```diff
@@ -50,6 +51,75 @@ func TestOCI1IndexFromManifest(t *testing.T) {
 	testValidManifestWithExtraFieldsIsRejected(t, parser, validManifest, []string{"config", "fsLayers", "history", "layers"})
 }
 
+func TestOCI1EditInstances(t *testing.T) {
+	validManifest, err := os.ReadFile(filepath.Join("testdata", "ociv1.image.index.json"))
+	require.NoError(t, err)
+	list, err := ListFromBlob(validManifest, GuessMIMEType(validManifest))
+	listSize := len(list.Instances())
+
+	require.NoError(t, err)
+	editInstances := []ListEdit{}
+	editInstances = append(editInstances, ListEdit{
+		UpdateOldDigest: list.Instances()[0],
+		UpdateDigest:    "sha256:bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
+		UpdateSize:      32,
+		UpdateMediaType: "something",
+		ListOperation:   ListOpUpdate})
+	err = list.EditInstances(editInstances)
+	require.NoError(t, err)
+
+	expectedDigests := list.Instances()
+	expectedDigests[0] = editInstances[0].UpdateDigest
+	// order of old elements must remain same.
+	assert.Equal(t, list.Instances(), expectedDigests)
+
+	instance, err := list.Instance(list.Instances()[0])
+	// We only edited the elements so list size must remain same.
+	assert.Equal(t, listSize, len(list.Instances()))
+	require.NoError(t, err)
```

Comment:
Please keep the `err` check immediately after the call that produces `err`; don’t interleave two checks.